runtime/src/balances.rs

```diff
@@ -414,6 +418,13 @@ decl_storage! {
 
         /// Any liquidity locks on some account balances.
         pub Locks get(locks): map T::AccountId => Vec<BalanceLock<T::Balance, T::BlockNumber>>;
+
+        /// Balance held by the identity. It can be spent by its signing keys.
```

Comment:
The comment `It can be spent by tis signing keys` is a bit confused after see `ChargeDid`.
Another question:  Is any key type/role able to spend this balance? I thought that specific key types and roles would be able to do that.

Reply:
In the current implementation, any signing key can spend an identity's balance on transaction fees. However, only the master key can transfer the balance back to itself.

The charge did variable is set by the signing keys to define if they want to use identity's balance or not.

Reply:
@adamdossa What roles and types of keys should be allowed to spend identity balance?

Reply:
I think a good model here is that the master key decided which of their signing keys is allowed to use identity funds to pay for transactions.

I think the neatest way to do this is via a role (i.e. a `KeyRole` of FUND is needed). We could add a roles parameter to `is_authorized_key(did: IdentityId, key: &Key, roles: Vec<KeyRole>)` and only have it return true if the `key` is a signing key of `did` and has every role in `roles` (which could be empty for no role checks).

We would use the same function for checking roles in any other extrinsics that require specific roles.